scipy/stats/tests/test_trapezoid_fix.py

```diff
@@ -0,0 +1,85 @@
+"""Test that trapezoid.fit now converges properly with default starting values."""
+
+import numpy as np
+import scipy.stats
+import pytest
+
+
+def test_trapezoid_fit_convergence():
+    """Test that trapezoid.fit converges without starting values."""
+    
+    # Generate test data from a trapezoidal distribution
+    true_c, true_d = 0.3, 0.7
+    true_dist = scipy.stats.trapezoid(true_c, true_d, 0, 1)
+    rvs = true_dist.rvs(1000, random_state=42)
+    
+    # Test 1: Fit without starting values (should now work)
+    fitted_args = scipy.stats.trapezoid.fit(rvs, floc=0, scale=1)
+    fitted_c, fitted_d = fitted_args[0], fitted_args[1]
+    
+    # Should not converge to triangular distribution (c=d=1)
+    assert not (abs(fitted_c - 1.0) < 0.01 and abs(fitted_d - 1.0) < 0.01), \
+        "fit() still converging to triangular distribution (c=d=1)"
+    
+    # Should converge to reasonable values
+    assert 0 <= fitted_c <= 1, f"fitted c={fitted_c} not in [0,1]"
+    assert 0 <= fitted_d <= 1, f"fitted d={fitted_d} not in [0,1]"
+    assert fitted_d >= fitted_c, f"fitted d={fitted_d} < c={fitted_c}"
```

Comment:
Can you make this test more rigorous by testing that the values are an expected value, instead of testing the properties that the values ought to have? I would suggest `assert_allclose` with an appropriate tolerance.

Given that this uses a seeded RNG and a deterministic optimizer, that should be fairly reliable.

Reply:
On a related note, the test will likely be more reliable if you use both floc and fscale.

Here loc is forced to be 0 but scale is set initially to 1 but allowed to vary.

```
fitted_args = scipy.stats.trapezoid.fit(rvs, floc=0, scale=1)
```